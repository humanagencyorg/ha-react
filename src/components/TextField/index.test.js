import React from 'react';
import renderer from 'react-test-renderer';

import { TextField } from './index';
import { mount } from 'enzyme';

describe('TextField component', () => {
  describe('without error', () => {
    it('matches snapshot', () => {
      const props = {
        name: 'name',
        label: 'label',
      };

      const wrapper = renderer.create(<TextField {...props} />);

      expect(wrapper.toJSON()).toMatchSnapshot();
    });
  });

  describe('with error', () => {
    it('matches snapshot', () => {
      const props = {
        name: 'name',
        label: 'label',
        error: true,
        helperText: 'error',
      };

      const wrapper = renderer.create(<TextField {...props} />);

      expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('shows error message', () => {
      const errorMessage = 'error';
      const props = {
        error: true,
        helperText: errorMessage,
      };

      const wrapper = mount(<TextField {...props} />);
      const errorWrapper = wrapper.find('p.Mui-error');

      expect(errorWrapper.text()).toBe(errorMessage);
    });
  });

  it('sets default value from props', () => {
    const defaultValue = 'value';
    const props = {
      defaultValue,
    };

    const wrapper = mount(<TextField {...props} />);
    const input = wrapper.find('input');

    expect(input.prop('defaultValue')).toBe(defaultValue);
  });

  it('calls #onFocus from props on field focus', () => {
    const onFocus = jest.fn();
    const props = {
      onFocus,
    };

    const wrapper = mount(<TextField {...props} />);
    const input = wrapper.find('input');

    input.simulate('focus');

    expect(onFocus).toHaveBeenCalled();
  });
});
