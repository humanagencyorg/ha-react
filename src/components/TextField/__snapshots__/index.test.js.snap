// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField component with error matches snapshot 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root makeStyles-root-1"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root makeStyles-label-3 MuiInputLabel-formControl MuiInputLabel-shrink MuiInputLabel-filled Mui-error Mui-error"
    data-shrink={true}
  >
    label
  </label>
  <div
    className="MuiInputBase-root MuiFilledInput-root Mui-error Mui-error MuiInputBase-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={true}
      autoFocus={false}
      className="MuiInputBase-input MuiFilledInput-input makeStyles-input-2 makeStyles-input-7"
      disabled={false}
      name="name"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
  </div>
  <p
    className="MuiFormHelperText-root makeStyles-errorMessage-4 MuiFormHelperText-contained Mui-error"
  >
    error
  </p>
</div>
`;

exports[`TextField component without error matches snapshot 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root makeStyles-root-1"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root makeStyles-label-3 MuiInputLabel-formControl MuiInputLabel-filled"
    data-shrink={false}
  >
    label
  </label>
  <div
    className="MuiInputBase-root MuiFilledInput-root MuiInputBase-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiFilledInput-input makeStyles-input-2 makeStyles-input-6"
      disabled={false}
      name="name"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
  </div>
</div>
`;
